sycl/plugins/opencl/pi_opencl.hpp

```diff
@@ -115,4 +224,33 @@ inline const OpenCLVersion V3_0(3, 0);
 
 } // namespace OCLV
 
+// Define the types that are opaque in pi.h in a manner suitable for OpenCL
+// plugin
+
+struct _pi_device : _pi_object {
+  enum device_level {
+    ROOTDEVICE = 0,
+    SUBDEVICE = 1,
+    SUBSUBDEVICE = 2,
+    INVALID = -1
+  };
+  _pi_device(pi_platform Plt) : Platform{Plt} {
+    level = INVALID;
+    family = index = 0;
+    // NOTE: one must additionally call initialize() to complete
```

Comment:
I don't see `_pi_device::initialize` here.

Reply:
Ah..the comment needs to be deleted.